flopy/utils/binaryfile.py

```diff
@@ -1343,7 +1346,8 @@ def _get_header(self):
         """
         header1 = binaryread(self.file, self.header1_dtype, (1,))
         nlay = header1["nlay"]
-        if nlay < 0:
+        self.compact = bool(nlay < 0)
```

Comment:
for the test to fail this must be false, which it seems could happen if 

- compact cbc files don't always have negative nlay (I thought they do, but could very well be wrong)
- something is up with the type/value of `nlay` &mdash; in my debugging with numpy >= 1.24 it is a 1D, 1-element ndarray, that was the motivation to wrap with `bool()`

Reply:
I was running numpy 1.23.5. I upgraded to numpy 1.26, but that did not solve the problem. It created a bunch of new error messages of the form shown below. So I am sorry to report it still doesn't work for me, and flopy needs some work for numpy 1.26 (for now adding a requirement that numpy < 1.26 until that is solved may be a good idea). 

```
/Users/markbakker/anaconda3/lib/python3.10/site-packages/flopy/utils/binaryfile.py:1692: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  ipos = int(self.iposarray[idx])
```

Reply:
Hey @mbakker7, I checked out @wpbonelli's fix and it works for me.  I debugged through it and saw it return the correct results.  I think what is happening is that you may have added his change to an out-of-date version of flopy:develop on your system.  The reason I say that is based on the error message that you posted above.  In that error message, it references `ipos = int(self.iposarray[idx])`, which was fixed in #1868 to support new numpy requirements.  Is that possible?

<img width="422" alt="image" src="https://github.com/modflowpy/flopy/assets/3372403/91c51c62-390b-4429-a007-d5c146ce97fd">

Reply:
You are right. I thought I was running the local version I have under git control on my machine, but I was running the master flopy version that I had installed. I should have realized that earlier. Sorry about wasting your time and thanks for the fix!

Reply:
@mbakker7 no problem! good news that there is no deeper issue here.